source/elements/oneMKL/source/domains/dft/config_params/data_layouts.rst

```diff
@@ -0,0 +1,299 @@
+considered along dimensions :math:`j \in \lbrace 1, \ldots, d\rbrace`.
+
+The default values set for the forward and backward strides correspond to the
+data layout configurations for unbatched, in-place transforms using unit stride
+along the last dimension with no offset (and minimal padding in forward
+domain in case of real descriptors, aligning with the :ref:`requirements for
+in-place transforms<onemkl_dft_data_layout_requirements>`). In other words, the
+default values are :math:`s^{\text{fwd}}_0 = s^{\text{bwd}}_0 = 0`,
+:math:`s^{\text{fwd}}_d = s^{\text{bwd}}_d = 1` and, for :math:`d`-dimensional
+DFTs with :math:`d > 1`,
+
+- :math:`s^{\text{fwd}}_{d-1} = s^{\text{bwd}}_{d-1} = n_{d}` for complex
+  descriptors;
+- :math:`s^{\text{bwd}}_{d-1} = \lfloor \frac{n_{d}}{2} \rfloor + 1`, and
+  :math:`s^{\text{fwd}}_{d-1} = 2 s^{\text{bwd}}_{d-1}` for real descriptors;
+- if :math:`d > 2`, :math:`s^{\text{xwd}}_k = n_{k+1} s^{\text{xwd}}_{k+1}`
+  for :math:`k \in \lbrace 1, \ldots, d - 2\rbrace` (for
+  :math:`\text{x} = \text{f}` and :math:`\text{x} = \text{b}`).
+
+.. _onemkl_dft_data_layout_requirements:
+
+.. rubric:: General consistency requirements
+
+In general, the distances and strides must be set so that every index value
+:eq:`eq_idx_data_layout` corresponds to a *unique* entry of the data sequences
+under consideration. In other words, there must not be one index value as
+expressed in :eq:`eq_idx_data_layout` that corresponds to two different
+:math:`(d+1)`-tuples :math:`(m, k_{1}, k_{2}, \dots, k_d)` that are both within
+the :ref:`elementary range of indices considered by
+oneMKL<onemkl_dft_elementary_range_of_indices>`.
+
+Additionally, for in-place transforms (configuration value
+``config_value::INPLACE`` associated with configuration parameter
+``config_param::PLACEMENT``), the smallest stride value must be associated with
+the same dimension in forward and backward domains and the data layouts must
+abide by following "*consistency requirement*": the memory address(es) of
+leading entry(ies) along the last dimension must be identical in forward and
+backward domains. Specifically, considering any :math:`(d+1)`-tuple
+:math:`(m, k_{1}, k_{2}, \dots, k_{d-1}, 0)` within :ref:`valid
+range<onemkl_dft_elementary_range_of_indices>`, the memory address of the
+element of corresponding index value :eq:`eq_idx_data_layout` in forward domain
+(considering the :ref:`implicitly assumed
+type<onemkl_dft_config_data_implicitly_assumed_elementary_data_type>` in forward
+domain) must be identical to the memory address of the element of corresponding
+index value :eq:`eq_idx_data_layout` in backward domain (considering the
+:ref:`implicitly assumed
+type<onemkl_dft_config_data_implicitly_assumed_elementary_data_type>` in
+backward domain). Equivalently,
+
+- for complex descriptors, the offset, stride(s) (and distances, if relevant)
+  must be equal in forward and backward domain;
```

Comment:
Relevant to another discussion here, we are already explicitly disallowing different input/output forward/backward strides for in-place transforms with this comment. So there's no change of behavior implied by removing the line about ignoring certain strides for in-place transforms.

Reply:
Well, not really... currently something like
`// desc is a fresh newly created 2D complex descriptor for sizes (n1 x n2), set for in-place by default`
`desc.set_value(config_param::INPUT_STRIDES, {0, n2, 1})`
`desc.commit(queue)`
is not expected to break since "output strides are ignored" due to the nature of the descriptor and (current) specified behavior for such in-place DFTs. With the changes herein though, the output strides (all of values 0) would not satisfy the consistency requirement and commit would be expected to fail.